### PR TITLE
CORE-538, CORE-571 Move local state to store

### PR DESCRIPTION
**Pull request by object-required**

### CODE REVIEW

File: src/components/interaction-designer/block-editors/GenericContactPropertyEditor.vue [hunk 1]
```diff
@@ -55,7 +55,7 @@
       <template v-if="flowSelectedContactPropertyField !== null">
         <label>{{ 'flow-builder.value' | trans }}</label>
         <div
-          v-if="isBlockInteractive(block)"
+          v-if="isBlockInteractive"
```
Comment (bulatgab) [inline src/components/interaction-designer/block-editors/GenericContactPropertyEditor.vue:58]: 👍

File: src/components/interaction-designer/block-editors/GroupMembershipEditor.vue [hunk 2]
```diff
@@ -93,7 +93,17 @@ export class GroupMembershipEditor extends mixins(Lang) {
...
+  set userAddedGroups(groups) {
+    this.block_updateVendorMetadataByPath({
+      blockId: this.block.uuid,
+      path: 'floip.ui_metadata.user_added_groups',
+      value: groups,
+    })
+  }
 
   get groupOptions(): IGroupMembership[] {
     return this.availableGroups ?? this.userAddedGroups
```
Comment (object-required) [inline src/components/interaction-designer/block-editors/GroupMembershipEditor.vue:109]: - [ ] Make sure `union()` works correctly for duplicates

**Comment by object-required** [comment]
- [ ] Clear `gropMembershipBlock.venvendor_metadata.floip.ui_metadata.user_added_groups` from `flow_persist` before saving the flow

File: package.json [hunk 3]
```diff
@@ -32,7 +32,7 @@
     "@avcs/autosuggest": "^1.7.5",
     "@floip/expression-evaluator": "^2.1.0",
     "@floip/expression-parser": "^1.11.3",
-    "@floip/flow-runner": "^1.0.12-rc4",
+    "@floip/flow-runner": "https://github.com/FLOIP/flow-runner.git#feature/CORE-538/CORE-571/move-local-state-to-store",
```
Comment (safydy-r) [inline package.json:35]: - [] publish flow-runner to nmp after we merge it, then consume it here

File: src/store/flow/utils/resourceHelpers.ts [hunk 4]
```diff
@@ -140,24 +141,36 @@ export function discoverContentTypesFor(mode: SupportedMode, resource?: IResourc
...
-export function cleanupFlowResources(container: IContext, choiceMimeType: string): IContext {
+export function cleanupFlowBeforePersisting(container: IContext, choiceMimeType: string): IContext {
+  const flows = structuredClone(container.flows)
+
+  flows.forEach((flow) => {
+    // Cleanup resources
+    flow.resources
+      .map((resource) => {
+        resource.values = resource.values.filter((value) => {
+          const isChoice = value.mime_type === choiceMimeType
```
Comment (safydy-r) [inline src/store/flow/utils/resourceHelpers.ts:152]: We can do an early return after `isChoice` ;)

**Review by safydy-r — APPROVED** [review]
No major concern, left some comments.
Pre-approving

File: package.json (hunk 3 above)
Comment (object-required) [reply package.json]: Made a note in https://github.com/FLOIP/flow-builder/pull/472, as it might be a bit early for releasing the package.

**Pull request merged**